pyfr/solvers/base/elements.py

```diff
@@ -87,8 +89,23 @@ def set_ics_from_cfg(self):
         if any(d in vars for d in 'xyz'):
             raise ValueError('Invalid constants (x, y, or z) in config file')
 
-        # Get the physical location of each solution point
-        coords = self.ploc_at_np('upts').swapaxes(0, 1)
+        # See if performing L2 projection
+        ename = self.basis.name
+        upts = self.cfg.get(f'solver-elements-{ename}', 'soln-pts')
+        qdeg = (self.cfg.getint('soln-ics', f'quad-deg-{ename}', 0) or 
+                self.cfg.getint('soln-ics', f'quad-deg', 0))
+        # Default to solution points if quad-pts are not specified
+        qpts = (self.cfg.get('soln-ics', f'quad-pts-{ename}', upts) or 
+                self.cfg.get('soln-ics', f'quad-pts', upts))
```

Comment:
I'd force the element type to always be specified.

Reply:
Just for the points or for the degree as well?

Reply:
Element type for sure, and I'd be inclined to go with degree, too but it might be worth seeing what the integrate plugin does here.

Reply:
Integrate uses a single option for all element quad degs, so I guess I'll go with that for consistency.

Reply:
Sounds good, we can change both if needed in the future.